src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -1042,4 +1061,492 @@ public void deleteHarvestedDataset(Dataset dataset, DataverseRequest request, Lo
+            Logger.getLogger(DatasetServiceBean.class.getName()).log(Level.SEVERE, null, ex);
+            return;
+        }
+
+        if (fileHandlerSuceeded) {
+            globusLogger.addHandler(fileHandler);
+        } else {
+            globusLogger = logger;
+        }
+
+        globusLogger.info("Starting an globusUpload ");
+
+        String datasetIdentifier = dataset.getStorageIdentifier();
+
+        String storageType = datasetIdentifier.substring(0, datasetIdentifier.indexOf("://") + 3);
+        datasetIdentifier = datasetIdentifier.substring(datasetIdentifier.indexOf("://") + 3);
+
+
+        Thread.sleep(5000);
+
+        JsonObject jsonObject = null;
+        try (StringReader rdr = new StringReader(jsonData)) {
+            jsonObject = Json.createReader(rdr).readObject();
+        } catch (Exception jpe) {
+            jpe.printStackTrace();
+            logger.log(Level.SEVERE, "Error parsing dataset json. Json: {0}");
+        }
+
+        String taskIdentifier = jsonObject.getString("taskIdentifier");
+
+        String ruleId = "";
+        try {
+            jsonObject.getString("ruleId");
+        } catch (NullPointerException npe) {
+
+        }
+
+        //  globus task status check
+        String taskStatus = globusStatusCheck(taskIdentifier, globusLogger);
+        Boolean taskSkippedFiles = taskSkippedFiles(taskIdentifier, globusLogger);
+
+        if(ruleId.length() > 0) {
+            globusServiceBean.deletePermision(ruleId, globusLogger);
+        }
+
+        if (taskStatus.startsWith("FAILED") || taskStatus.startsWith("INACTIVE")) {
+            String comment = "Reason : " + taskStatus.split("#") [1] + "<br> Short Description : " + taskStatus.split("#")[2];
+            userNotificationService.sendNotification((AuthenticatedUser) authUser, new Timestamp(new Date().getTime()), UserNotification.Type.GLOBUSUPLOADCOMPLETEDWITHERRORS, dataset.getId(),comment, true);
+            globusLogger.info("Globus task failed ");
+
+            DatasetLock dcmLock = dataset.getLockFor(DatasetLock.Reason.EditInProgress);
+            if (dcmLock == null) {
+                logger.log(Level.WARNING, "No lock found for dataset");
+            } else {
+                removeDatasetLocks(dataset, DatasetLock.Reason.EditInProgress);
+                logger.log(Level.INFO, "Removed EditInProgress lock ");
+                //dataset.removeLock(dcmLock);
+            }
+        }
+        else {
+            try {
+                List<String> inputList = new ArrayList<String>();
+                JsonArray filesJsonArray = jsonObject.getJsonArray("files");
+
+                if (filesJsonArray != null) {
+
+                    for (JsonObject fileJsonObject : filesJsonArray.getValuesAs(JsonObject.class)) {
+
+                        //   storageIdentifier   s3://gcs5-bucket1:1781cfeb8a7-748c270a227c from externalTool
+                        String storageIdentifier = fileJsonObject.getString("storageIdentifier");
```

Comment:
I'm not sure if there's all the functionality you need but there are methods such as DataAcess.getDriverAndStorageLocation() and getStorarageIdFromLocation() (sic - misspelled in the code) that can do some of the manipulation you need.